### PR TITLE
add metadata support

### LINKED ISSUE #622: Feature: metadata support
### Feature description

S3 supports setting metadata for each file.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html

My narrow usecase is to make a pipeline from S3 to [openai vector stores](https://platform.openai.com/docs/api-reference/vector-stores) possible. Each uploaded file to openai must have a [purpose](https://platform.openai.com/docs/api-reference/files/create#files-create-purpose), so in this specific case, the purpose of the file could be stored in the S3 file metadata.

It should also be useful in other usecases for sure.

### PR DESCRIPTION
**Pull request by alarbada**
### Description

Adds metadata support for the S3 source connector.
Fixes #622 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-s3/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Comment by alarbada** [comment]
(Note to the future reviewer): I noticed that there's a bit of annoying duplication in how we handle env secrets. Ideally both source and destination should share the same way of fetching the secrets from the environment, so that the setup is a bit less stringly typed.

We could do the same way we do it in the [mysql connector](https://github.com/conduitio-labs/conduit-connector-mysql), have a testutils package.

What do you think?

File: source/iterator/cdc_iterator.go [hunk 1]
```diff
@@ -239,30 +239,32 @@ func (w *CDCIterator) buildRecord(entry CacheEntry) (opencdc.Record, error) {
 		Type:      position.TypeCDC,
 	}
 
+	m := opencdc.Metadata{}
+	if object != nil {
+		m[MetadataContentType] = *object.ContentType
+		for key, val := range object.Metadata {
```
Comment (hariso) [inline source/iterator/cdc_iterator.go:245]: In some other connectors, we've also been adding a prefix that shows from which connector did the metadata come from, e.g. **[here](https://github.com/ConduitIO/conduit-connector-kafka/blob/main/source.go#L110).**

File: source/source_integration_test.go [hunk 2]
```diff
@@ -644,8 +645,21 @@ func createTestBucket(t *testing.T, client *s3.Client, bucket string) {
 		t.Logf("created test bucket %q in %v", bucket, time.Since(start))
 	}()
 
+	// this allows the integration tests to run in buckets that are not in us-east-1
+	region := os.Getenv("AWS_REGION")
+	var createConfig *types.CreateBucketConfiguration
+	for _, value := range types.BucketLocationConstraint.Values("") {
```
Comment (hariso) [inline source/source_integration_test.go:651]: What's this part about?

File: source/iterator/cdc_iterator.go (hunk 1 above)
Comment (alarbada) [reply source/iterator/cdc_iterator.go:245]: Nice idea, pushed new changes.

File: source/source_integration_test.go (hunk 2 above)
Comment (alarbada) [reply source/source_integration_test.go:651]: My aws account did put my s3 into frankfurt for some reason. When I tried to run the integration tests with my account credentials I got a bucket creation error, something about requiring a location constraint. This did fix it.

**Review by hariso — APPROVED** [review]

**Comment by hariso** [comment]
> (Note to the future reviewer): I noticed that there's a bit of annoying duplication in how we handle env secrets. Ideally both source and destination should share the same way of fetching the secrets from the environment, so that the setup is a bit less stringly typed.
> 
> We could do the same way we do it in the [mysql connector](https://github.com/conduitio-labs/conduit-connector-mysql), have a testutils package.
> 
> What do you think?

Hmm, can you maybe show an example from this connector? I'm not sure I understand what exactly is the problem with env. secrets.

File: source/source_integration_test.go (hunk 2 above)
Comment (alarbada) [reply source/source_integration_test.go:651]: Just added better comment! At the end was more about the test s3 client using an aws region different than the default one, `us-east-1`. Should be a bit easier to understand now.

**Comment by hariso** [comment]
@alarbada LGTM! Btw, can you clarify your comment about env. secrets above?

**Comment by alarbada** [comment]
> @alarbada LGTM! Btw, can you clarify your comment about env. secrets above?

Yeah @hariso, it's a pattern I tend to use now in new connectors that have multiple packages. Most of the times the setup that I use to start a source connector is very similar to the destination connector. I don't mind usually repeating a bit of code, but it helps having a consistent way of setting up the connector.

In this repo for example you can see that we have a `parseIntegrationConfig()` in the source connector to setup the configuration, but in the destination connector we've got the `getEnv` utility. I believe that by just putting all that code into a single shared testutils  makes it easier to have a more consistent way of testing the connector.

I did that already for the kinesis and the mysql connector, and it is definitely a pattern I recommend for testing connectors.

**Pull request merged**